upf/pfcp.c

```diff
@@ -861,6 +857,12 @@ encode_f_teid (void *p, u8 ** vec)
   pfcp_f_teid_t *v = p;
 
   put_u8 (*vec, v->flags);
+
+  //CH and CHID are only set by CP function.
+  if ((v->flags & F_TEID_CH) || (v->flags & F_TEID_CH))
+    return -1;
+
+
```

Comment:
the PFCP en/decoder has no notion of direction or side, and this is how it was intended. Checks like that belong in the code using it.